src/utils/userLocation.ts

```diff
@@ -55,26 +86,32 @@ export const enterExternalRoom = ({ userId, room }: EnterExternalRoomProps) => {
 };
 
 export const useUpdateTimespentPeriodically = (
-  user: UserInfo | undefined,
-  roomName: string
+  roomName: string,
+  userId?: string
 ) => {
```

Comment:
Why was the position of these parameters changed? Guessing because TypeScript wanted the optional param to be last? We should probably change this to use a 'props object' with object destructuring + named keys instead of positional parameters.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1129/commits/53f57c40e24983470b7552d4ebed6f1715ff74ba